File: src/main/java/com/ab/ploy/filters/PloyStreamWrapper.java
package com.ab.ploy.filters;

import java.io.IOException;
import java.io.InputStream;
import javax.servlet.ReadListener;
import javax.servlet.ServletInputStream;

public class PloyStreamWrapper extends ServletInputStream {

    private final InputStream oldStream;
    private final InputStream newStream;

    public PloyStreamWrapper(InputStream oldStream, InputStream newStream) {
        this.oldStream = oldStream;
        this.newStream = newStream;
    }

    @Override
    public void close() throws IOException {
        oldStream.close();
        newStream.close();
    }

    @Override
    public boolean isFinished() {
        return true;
    }

    @Override
    public boolean isReady() {
        return true;
    }

    @Override
    public void setReadListener(ReadListener listener) {
    }

    @Override
    public int read() throws IOException {
        return newStream.read();
    }
}

File: src/main/kotlin/com/ab/ploy/common/filters/PloyStreamWrapper.kt
/* Copyright © 2021 */
package com.ab.ploy.common.filters

import java.io.InputStream
import javax.servlet.ReadListener
import javax.servlet.ServletInputStream

class PloyStreamWrapper(private val oldStream: InputStream, private val newStream: InputStream) :
    ServletInputStream() {
    override fun read(): Int {
        return newStream.read()
    }

    override fun close() {
        oldStream.close()
        newStream.close()
    }

    override fun isFinished(): Boolean {
        return true
    }

    override fun isReady(): Boolean {
        return true
    }

    override fun setReadListener(rl: ReadListener?) {}
}
